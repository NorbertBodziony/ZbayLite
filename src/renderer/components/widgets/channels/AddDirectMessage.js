import React from 'react'
import * as R from 'ramda'

import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'

import AddIcon from '@material-ui/icons/Add'

import MenuAction from '../../ui/MenuAction'

const styles = theme => ({
  icon: {
    fontSize: 26
  },
  button: {
    padding: 4,
    color: theme.typography.body2.color
  }
})

export const AddDirectMessage = ({ classes, openModal }) => {
  return (
    <React.Fragment>
      <MenuAction
        classes={{
          button: classes.button,
          icon: classes.icon
        }}
        Icon={AddIcon}
        IconButton={IconButton}
        offset='0 8'
        onClick={openModal}
      />
    </React.Fragment>
  )
}

export default R.compose(
  withStyles(styles),
  React.memo
)(AddDirectMessage)
