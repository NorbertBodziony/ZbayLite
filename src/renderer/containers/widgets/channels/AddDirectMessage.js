import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import AddDirectMessage from '../../../components/widgets/channels/AddDirectMessage'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch => bindActionCreators({
  openModal: actionCreators.openModal('sendMoney')
}, dispatch)

export default connect(null, mapDispatchToProps)(AddDirectMessage)
