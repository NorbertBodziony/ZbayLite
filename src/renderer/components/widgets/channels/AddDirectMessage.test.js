import React from 'react'
import { shallow } from 'enzyme'

import { AddDirectMessage } from './AddDirectMessage'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('AddDirectMessage', () => {
  it('renders component', () => {
    const result = shallow(<AddDirectMessage classes={mockClasses} />)
    expect(result).toMatchSnapshot()
  })
})
