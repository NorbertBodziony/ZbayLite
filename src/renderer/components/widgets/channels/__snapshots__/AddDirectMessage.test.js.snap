// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDirectMessage renders component 1`] = `
<Fragment>
  <WithStyles(MenuAction)
    Icon={[Function]}
    IconButton={[Function]}
    classes={
      Object {
        "button": "button-mock",
        "icon": "icon-mock",
      }
    }
    offset="0 8"
  />
</Fragment>
`;
